File: completable-reactor-graph/src/main/java/ru/fix/completable/reactor/graph/MergerBuilder.java
package ru.fix.completable.reactor.graph;

public interface MergerBuilder<Payload, HandlerResult> {

    Vertex withMerger(Merger<Payload, HandlerResult> merger);

    Vertex withRoutingMerger(RoutingMerger<Payload, HandlerResult> merger);

    Vertex withEmptyMerger();

    Vertex withoutMerger();
}

File: completable-reactor-graph/src/main/kotlin/ru/fix/completable/reactor/graph/MergerBuilder.kt
package ru.fix.completable.reactor.graph

interface MergerBuilder<Payload, HandlerResult> {

    fun withMerger(merger: Merger<Payload, HandlerResult>): Vertex

    //TODO: add javadoc for all vertices
    fun withRoutingMerger(merger: RoutingMerger<Payload, HandlerResult>): Vertex

    //TODO: specialize Vertices: router, merger, so handleBy and mergeBy will not conflict for router
    fun withEmptyMerger(): Vertex

    fun withoutMerger(): Vertex
}
